### DISCUSSION TITLE
WASM, navigationview using routemap doesn't seems to set ViewModel

### DISCUSSION THREAD

**Discussion by digger73** [unoplatform/uno.extensions#1669] [Q&A]
Hi

I'm doing some simple app to get to know UNO.
Currently I'm using NavigationView and have set up a RouteMap in App.cs that looks like this:
`new RouteMap("OnlineValues", View: views.FindByViewModel<OnlineValuesViewModel>()),`
It works, as navigation to the page works fine.

I have furthermore registered a view between the view and ViewModel.

Here is the problem. On the OnlineValuesPage I have created, I have a simple ListView and also a DataGrid, that both binds to the same observableCollection in the ViewModel. This works fine in native windows app and in Android. Initial values are shown and updates works fine.
However, in the WASM project the initial values added to the collection in the constructor are not shown. Furthermore, it seems that the commands I have on two buttons doesn't hit the breakpoints when clicking the buttons.

I'm using normal binding, tried to setup a viewModel property in the code behind for the view to use x:bind, but this results in the page not loading at all when navigating to it in WASM, other projects works.

I have looked around in the docs and the internet to see if something special have to be done to get the bindings to work. Does anyone have any clue to get it to work ?

**Comment by nickrandolph** [comment]
Which version of Uno.Extensions and Uno are you using - there was an issue with earlier versions of uno.extensions where x:Bind wasn't working because the viewmodel was set after the initial bindings were evaluated. This should be fixed in the most recent stable release though.

**Reply by digger73** [reply]
I have just update everything in the project, still doesn't work with the bindings. I will make a small sample project to show what the problem is.

**Reply by digger73** [reply]
Well it started working after removing underlying business code so know I guess I just have to figure out what part of it that is the cause...

**Reply by digger73** [reply]
Found the reason for it not working. I have a DataAccess class that is initialized through DI, in the constructor a GrpcChannel is made (I do know it isn't supported in web). No warnings or any indication in the outputs that this broke the initialization of the viewmodel in the later steps of the service handling the DI.

**Reply by nickrandolph** [reply]
If you set logging to Trace, do you see any errors/warnings in the logs. We use the Microsoft DI container, so if a viewmodel fails to initialize, it should generate an error in the log

**Reply by digger73** [reply]
I tried that, but I can't see anything obvious errors. 
It seems however to be due to an exception thrown in the constructor, so I moved out the code to be called when a button is pushed, meaning that the viewModel is correctly initialized and binding is working. 
However, an error still occurs when calling the method, but I guess that requires another question, related to that specific problem.
Thanks for your time :)

**Comment by digger73** [comment]
Hi 
Thanks for the suggestion, I'm not using x:bind as that becomes a problem when trying to set the viewModel in the code behind. So I'm using normal binding using DI and route etc. where it seems not to set the dataContext to the viewModel as it does for the other platforms.
I'm using uno.tookit.winui 3.0.4, Uno.WinUI 4.9.20 and UNO.extension 2.4.2 which seems to be the newest.
